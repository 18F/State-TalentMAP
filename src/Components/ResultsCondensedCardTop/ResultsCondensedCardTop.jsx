import React from 'react';
import { POSITION_DETAILS } from '../../Constants/PropTypes';

const ResultsCondensedCardTop = ({ position }) => (
  <div className="usa-grid-full condensed-card-top">
    <div className="usa-grid-full condensed-card-top-header-container">
      <div
        className={
          'usa-width-two-thirds condensed-card-top-header condensed-card-top-header-left'
        }
      >
        {position.title}
      </div>
      <div
        className="usa-width-one-third condensed-card-top-header condensed-card-top-header-right"
      >
        Grade: {position.grade}
      </div>
    </div>
  </div>
);

ResultsCondensedCardTop.propTypes = {
  position: POSITION_DETAILS.isRequired,
};

export default ResultsCondensedCardTop;
