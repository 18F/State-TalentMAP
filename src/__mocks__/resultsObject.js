const resultsObject = {
  count: 2,
  results: [
    { id: 6,
      grade: '05',
      skill: 'OFFICE MANAGEMENT (9017)',
      bureau: '150000',
      organization: 'YAOUNDE CAMEROON (YAOUNDE)',
      position_number: '00025003',
      is_overseas: true,
      create_date: '2006-09-20',
      update_date: '2017-06-08',
      bid_statistics: [
        {
          id: 1,
          total_bids: 3,
          in_grade: 0,
          at_skill: 0,
          in_grade_at_skill: 0,
          bidcycle: 'Demo BidCycle 2018-01-12 15:11:38.004793',
        },
      ],
      description: {
        content: 'The quick brown fox jumped over the lazy dog.',
        date_created: '2018-01-12T15:09:22.952226Z',
      },
      post: { id: 162, obc_id: 162, tour_of_duty: '2YRR', code: 'LT6000000', location: 'MASERU, LESOTHO', cost_of_living_adjustment: 0, differential_rate: 15, danger_pay: 0, rest_relaxation_point: 'London', has_consumable_allowance: false, has_service_needs_differential: false },
      languages: [
        { id: 1, language: 'French (FR)', reading_proficiency: '2', spoken_proficiency: '2', representation: 'French (FR) 2/2' },
        { id: 1, language: 'German (GM)', reading_proficiency: '2', spoken_proficiency: '2', representation: 'German (GM) 2/2' },
      ],
      title: 'Position - First',
    },
    { id: 60,
      grade: '03',
      skill: 'OFFICE MANAGEMENT (9017)',
      bureau: '150000',
      organization: 'YAOUNDE CAMEROON (YAOUNDE)',
      position_number: '00025003',
      is_overseas: true,
      create_date: '2006-09-20',
      update_date: '2017-06-08',
      bid_statistics: [
        {
          id: 1,
          total_bids: 3,
          in_grade: 0,
          at_skill: 0,
          in_grade_at_skill: 0,
          bidcycle: 'Demo BidCycle 2018-01-12 15:11:38.004793',
        },
      ],
      post: null,
      languages: [],
      title: 'Position - Second',
    },
  ],
};

export default resultsObject;
