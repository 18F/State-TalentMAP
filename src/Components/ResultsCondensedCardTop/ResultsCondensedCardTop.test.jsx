import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import ResultsCondensedCardTop from './ResultsCondensedCardTop';
import resultsObject from '../../__mocks__/resultsObject';

describe('ResultsCondensedCardTopComponent', () => {
  const type = 'new';
  it('is defined', () => {
    const wrapper = shallow(
      <ResultsCondensedCardTop
        position={resultsObject.results[0]}
        type={type}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('can receive props', () => {
    const wrapper = shallow(
      <ResultsCondensedCardTop
        position={resultsObject.results[0]}
        type={type}
      />,
    );
    expect(wrapper.instance().props.type).toBe(type);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <ResultsCondensedCardTop
        position={resultsObject.results[0]}
        type={type}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
