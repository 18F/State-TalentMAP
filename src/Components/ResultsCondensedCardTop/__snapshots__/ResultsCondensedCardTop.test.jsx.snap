// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsCondensedCardTopComponent matches snapshot 1`] = `
<div
  className="usa-grid-full condensed-card-top"
>
  <div
    className="usa-grid-full condensed-card-top-header-container"
  >
    <div
      className="usa-width-two-thirds condensed-card-top-header condensed-card-top-header-left"
    />
    <div
      className="usa-width-one-third condensed-card-top-header condensed-card-top-header-right"
    >
      Grade: 
      05
    </div>
  </div>
</div>
`;
